### PR TITLE
[MM-25763] show email invite errors

### PR DESCRIPTION
**Pull request by Willyfrog**
#### Summary

Make errors while sending email invites more visible

#### Ticket Link

[MM-25763](https://mattermost.atlassian.net/browse/MM-25763)

#### Additional notes

this PR contains *breaking changes* to the API as invites might now return new types of errors (originally only permission errors were expected)
#### Release Note

```release-note
Added a 500 error if something goes wrong while inviting to a team, this also enables the UI to be aware that not all invites were sent.
```

[MM-25763]: https://mattermost.atlassian.net/browse/MM-25763?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: i18n/en.json
Comment (Willyfrog) [inline i18n/en.json:4356]: ci is failing on this, but running `make i18n-extract` decides to remove this. any ideas what's going on?

Comment (Willyfrog) [reply i18n/en.json]: nevermind, had the enterprise repo a bit out of date. updating and running again allowed to fix

File: app/team.go
Comment (isacikgoz) [inline app/team.go:1152]:
```suggestion
```

File: app/team.go
Comment (isacikgoz) [inline app/team.go:1156]:
```suggestion
	return inviteListWithErrors, nil
```

**Review by isacikgoz — DISMISSED** [review]
Looks good, have a couple of non-blocking suggestions below 👇

File: app/team.go
Comment (mgdelacroix) [inline app/team.go:1273]: nit: can be simplified to

```go
 	return a.SendInviteEmails(team, user.GetDisplayName(nameFormat), user.Id, emailList, a.GetSiteURL()) 
```

**Comment by Willyfrog** [comment]
added review comments and the guest functionality

**Comment by mgdelacroix** [comment]
Stepping down as a reviewer because of holidays

File: app/team.go
Comment (ethervoid) [inline app/team.go:1309]: One thing that we should be careful with is that in the past we didn't return errors and now we're doing it so maybe that would end-up failing when right now we're not failing. Are we sure that returning the error is not going to provoke problems in the public API?

File: app/team.go
Comment (ethervoid) [inline app/team.go:1341]: Same here

File: app/email.go
Comment (isacikgoz) [inline app/email.go:389]: I think `422` is a little bit misleading here. We can parse errors of `es.sendMail` or we can pick the shortest path:

```suggestion
		return model.NewAppError("emailSentFailed", "api.user.invite.error", nil, strings.Join(errorMails, ", "), http.StatusInternalServerError)
```

File: app/email.go
Comment (isacikgoz) [inline app/email.go:486]: Same as above.

File: app/email.go
Comment (Willyfrog) [reply app/email.go:389]: i chose [unprocessable entity](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) as some might fail but not all, so I didn't want to imply that all has go wrong or something failed on the server, it is just that something is not right with some section of the request. 

Anyways I'm 0/5 on this and I already expected it might be preferred to have a 500 error, so i'll change that.

File: app/team.go
Comment (Willyfrog) [reply app/team.go:1309]: good question, but I can only guess!
doing a search in the [webapp repo](https://github.com/mattermost/mattermost-webapp/search?q=sendEmailInvitesToTeamGracefully&unscoped_q=sendEmailInvitesToTeamGracefully) only returns two results. one is within a `try` block and the other is a test. So initially I'd say it wont, but there might be plugins using that feature that might fail or not expect such response.
i will submit a PR for the api docs and add a note in the PR about the breaking changes, but I'm not aware of any other means of comunicating/checking it.

**Comment by Willyfrog** [comment]
Added tests and other review requests. @ethervoid & @isacikgoz can you re-review?
in the meantime this is blocked until platform team decides if this is a breaking change or not, and what needs to be done.

**Comment by emilyacook** [comment]
@Willyfrog let me know if there's any action that can be taken on this so I can set the status per our [inactive contribution guidelines](https://developers.mattermost.com/contribute/getting-started/inactive-contributions/)

**Comment by Willyfrog** [comment]
@emilyacook no, it's on my end, I have to fix the review comments on the other one before merging this one
I'll try get that done next week

**Comment by emilyacook** [comment]
My bad @Willyfrog didnt realize staff PRs were also getting the Stale label, it was my first week doing this too smh sorry 😅

**Comment by isacikgoz** [comment]
Hi @Willyfrog, Note that we [merged multiple repositories into one monorepo](https://github.com/mattermost/mattermost-server/issues/22420) which may require you to resubmit these changes. I'm closing in the favor of reducing load on mattermod. Also the CI checks should run on our new pipelines. You can update the branch and re-open if you feel necessary.

**Pull request closed without merging**
